xbmc/cores/VideoPlayer/DVDInputStreams/InputStreamAddon.cpp

```diff
@@ -237,29 +357,129 @@ bool CInputStreamAddon::SeekTime(double time, bool backward, double* startpts)
+}
+
+bool CInputStreamAddon::IsRealTimeStream()
+{
+  if (m_struct.toAddon.is_real_time_stream)
+    return m_struct.toAddon.is_real_time_stream(&m_struct);
+  return false;
+}
+
+void CInputStreamAddon::UpdateStreams()
+{
+  DisposeStreams();
+
+  INPUTSTREAM_IDS streamIDs = m_struct.toAddon.get_stream_ids(&m_struct);
+  if (streamIDs.m_streamCount > INPUTSTREAM_IDS::MAX_STREAM_COUNT)
+  {
+    DisposeStreams();
     return;
+  }
 
-  m_addon->SetVideoResolution(width, height);
+  for (unsigned int i = 0; i < streamIDs.m_streamCount; ++i)
+  {
+    INPUTSTREAM_INFO stream = m_struct.toAddon.get_stream(&m_struct, streamIDs.m_streamIds[i]);
+    if (stream.m_streamType == INPUTSTREAM_INFO::TYPE_NONE)
+      continue;
+
+    std::string codecName(stream.m_codecName);
+    StringUtils::ToLower(codecName);
+    AVCodec *codec = avcodec_find_decoder_by_name(codecName.c_str());
+    if (!codec)
+      continue;
+
+    CDemuxStream *demuxStream;
+
+    if (stream.m_streamType == INPUTSTREAM_INFO::TYPE_AUDIO)
+    {
+      CDemuxStreamAudio *audioStream = new CDemuxStreamAudio();
+
+      audioStream->iChannels = stream.m_Channels;
+      audioStream->iSampleRate = stream.m_SampleRate;
+      audioStream->iBlockAlign = stream.m_BlockAlign;
+      audioStream->iBitRate = stream.m_BitRate;
+      audioStream->iBitsPerSample = stream.m_BitsPerSample;
+      demuxStream = audioStream;
+    }
+    else if (stream.m_streamType == INPUTSTREAM_INFO::TYPE_VIDEO)
+    {
+      CDemuxStreamVideo *videoStream = new CDemuxStreamVideo();
+
+      videoStream->iFpsScale = stream.m_FpsScale;
+      videoStream->iFpsRate = stream.m_FpsRate;
+      videoStream->iWidth = stream.m_Width;
+      videoStream->iHeight = stream.m_Height;
+      videoStream->fAspect = stream.m_Aspect;
+      videoStream->stereo_mode = "mono";
+      videoStream->iBitRate = stream.m_BitRate;
+      demuxStream = videoStream;
+    }
+    else if (stream.m_streamType == INPUTSTREAM_INFO::TYPE_SUBTITLE)
+    {
+      CDemuxStreamSubtitle *subtitleStream = new CDemuxStreamSubtitle();
+      demuxStream = subtitleStream;
+    }
+    else
+      continue;
+
+    demuxStream->codec = codec->id;
+    demuxStream->codecName = stream.m_codecInternalName;
+    demuxStream->uniqueId = streamIDs.m_streamIds[i];
+    demuxStream->language[0] = stream.m_language[0];
+    demuxStream->language[1] = stream.m_language[1];
+    demuxStream->language[2] = stream.m_language[2];
+    demuxStream->language[3] = stream.m_language[3];
+
+    if (stream.m_ExtraData && stream.m_ExtraSize)
+    {
+      demuxStream->ExtraData = new uint8_t[stream.m_ExtraSize];
+      demuxStream->ExtraSize = stream.m_ExtraSize;
+      for (unsigned int j = 0; j < stream.m_ExtraSize; ++j)
+        demuxStream->ExtraData[j] = stream.m_ExtraData[j];
+    }
+
+    m_streams[demuxStream->uniqueId] = demuxStream;
+  }
+}
+
+void CInputStreamAddon::DisposeStreams()
+{
+  for (auto &stream : m_streams)
+    delete stream.second;
```

Comment:
does the dtor of stream.second take care to delete `ExtraData`, which is allocated here https://github.com/xbmc/xbmc/pull/12325/files#diff-3590db0d5bee056ad4a07281388f0531R455?